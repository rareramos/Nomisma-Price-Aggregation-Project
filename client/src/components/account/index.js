export { Account as default} from './account-container';
