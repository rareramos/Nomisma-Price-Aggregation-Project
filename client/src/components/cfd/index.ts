export { Cfd } from './cfd-container';
