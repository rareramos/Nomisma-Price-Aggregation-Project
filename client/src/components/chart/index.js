export { Chart } from './chart-container';
