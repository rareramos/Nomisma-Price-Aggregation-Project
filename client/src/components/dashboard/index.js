export { Dashboard } from './dashboard-container';
