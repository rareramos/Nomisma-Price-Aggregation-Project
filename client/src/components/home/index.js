export { Home as default } from './home-container';
