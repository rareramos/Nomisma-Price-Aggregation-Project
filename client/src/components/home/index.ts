export { Home } from './home-container';
