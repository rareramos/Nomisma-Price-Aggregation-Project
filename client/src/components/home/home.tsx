import React, { Component, ReactNode } from 'react';
import { IHomeProps } from 'types/components/home';
import { Cfd } from '../cfd';
import { CfdWrapper } from './styled';

export class Home extends Component<IHomeProps> {
  public componentDidMount() : void {
    const { navigateToCFD } = this.props;
    navigateToCFD('cfd');
  }

  public render() : ReactNode {
    return (
      <CfdWrapper>
        <Cfd />
      </CfdWrapper>
    );
  }
}
