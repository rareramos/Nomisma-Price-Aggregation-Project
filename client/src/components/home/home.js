import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Loading } from '@nomisma/nomisma-ui/beta/loading';
import { IconSortUp, IconSortDown } from '@nomisma/nomisma-ui/icons';
import { Table } from 'reactstrap';
import Pagination from 'react-js-pagination';

import { Styled } from './styled';
import { TableToolbar } from '../table-toolbar';
import { Chart } from '../chart';

export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activePage: 1,
    };
    this.PageChange = this.PageChange.bind(this);
  }
  PageChange(pagenumber) {
    this.setState({ activePage: pagenumber });
    this.props.handlePageChange(pagenumber);
  }

  renderSortIcon(order) {
    const style = { marginLeft: '0.25rem', fill: 'var(--primary)' };
    if (order === 1) {
      return <IconSortUp style={ style }/>;
    } else if (order === -1) {
      return <IconSortDown style={ style }/>;
    }
    return null;
  }

  render() {
    const {
      loading,
      tableData,
      pageCount,
      columns,
      sortByColumn,
      filter,
    } = this.props;
    const itemCounts = (pageCount - 1) * 15;
    return (
      <Styled>
        <Chart/>
        <TableToolbar/>
        {
          /* eslint-disable-next-line no-nested-ternary */
          loading ? <Loading fill='var(--primary)' /> : tableData.length > 0 ? (
            <Fragment>
              <Table>
                <thead>
                  <tr>
                    { columns.map(({ key, columnName }) => (
                      <th
                        key={ columnName }
                        onClick={ sortByColumn.bind(null, key) }
                      >
                        { columnName }
                        { filter.sort === key ? this.renderSortIcon(filter.order) : null }
                      </th>
                    )) }
                  </tr>
                </thead>
                <tbody>
                  { tableData.map(
                    (
                      item,
                      idx,
                    ) =>
                      (
                        <tr key={ idx }>
                          { columns.map(
                            (
                              {
                                key,
                              }
                            ) => (
                              <td
                                key={ `${key}-${idx}` }
                              > { item[key] } </td>
                            )
                          )
                          }
                        </tr>
                      )
                  )
                  }
                </tbody>
              </Table>
              <Pagination
                activePage={ this.state.activePage }
                itemsCountPerPage={ 15 }
                totalItemsCount={ itemCounts }
                pageRangeDisplayed={ 5 }
                onChange={ this.PageChange }
              />
            </Fragment>
          ) : <p>No results found...</p>
        }
      </Styled>
    );
  }
}

Home.propTypes = {
  loading: PropTypes.bool.isRequired,
  pageCount: PropTypes.number.isRequired,
  tableData: PropTypes.arrayOf(PropTypes.object),
  handlePageChange: PropTypes.func.isRequired,
  columns: PropTypes.array.isRequired,
  sortByColumn: PropTypes.func.isRequired,
  filter: PropTypes.object.isRequired,
};
