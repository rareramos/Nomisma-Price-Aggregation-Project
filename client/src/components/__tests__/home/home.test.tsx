import * as React from 'react';
import { shallow } from 'enzyme';

import { IHomeProps } from 'types/components/home';
import { Home } from 'components/home/home';

const HomeProps : IHomeProps = {
  navigateToCFD: jest.fn,
};

describe('Home test', () => {
  it('matches home snapshot', () => {
    const wrapper = shallow(<Home {...HomeProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
