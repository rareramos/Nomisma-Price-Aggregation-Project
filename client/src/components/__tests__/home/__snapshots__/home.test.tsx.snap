// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test matches home snapshot 1`] = `
<styled.div>
  <Connect(Cfd) />
</styled.div>
`;
